File: chtl/src/chtl_compiler/loader/loader.rs
use std::fs;
use std::io;
use std::path::Path;

pub struct Loader;

impl Loader {
    pub fn new() -> Self {
        Loader
    }

    pub fn load_file(&self, path: &str) -> io::Result<String> {
        fs::read_to_string(Path::new(path))
    }
}

File: Loader/Loader.cpp
#include "Loader.h"
#include <fstream>
#include <sstream>
#include <iostream>

std::string Loader::readTextFile(const std::string& path) {
    // For now, assume path is relative to the execution directory.
    // The spec requires a more complex search path logic.
    std::ifstream file(path);
    if (!file.is_open()) {
        // In a real compiler, we might want to handle this more gracefully.
        std::cerr << "Warning: Could not open imported file: " << path << std::endl;
        return "";
    }
    std::stringstream buffer;
    buffer << file.rdbuf();
    return buffer.str();
}
